core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -67,24 +82,65 @@ oppia.directive('topNavigationBar', [
             $window.localStorage.removeItem('last_uploaded_audio_lang');
           };
 
-          $scope.profileDropdownIsActive = false;
-          $scope.onMouseoverProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
-            $scope.profileDropdownIsActive = true;
-          };
-          $scope.onMouseoutProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
-            $scope.profileDropdownIsActive = false;
+          /**
+           * Opens the submenu.
+           * @param {object} evt
+           * @param {String} menuName - name of menu, on which
+           * open/close action to be performed (aboutMenu,profileMenu).
+           */
+          $scope.openSubmenu = function(evt, menuName) {
+            // Focus on the current target before opening its submenu.
+            angular.element(evt.currentTarget).focus();
+            $scope.activeMenuName = menuName;
           };
-          $scope.onMouseoutDropdownMenuAbout = function(evt) {
-            angular.element(evt.currentTarget)[0].blur();
+          $scope.blurNavigationLinks = function(evt) {
+            // This is required because if about submenu is in open state
+            // and when you hover on library, both will be highlighted,
+            // To avoid that, blur all the a's in nav, so that only one 
+            // will be highlighted.
+            $('nav a').blur();
           };
-          $scope.onMouseoverDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
+          $scope.closeSubmenu = function(evt) {
+            $scope.activeMenuName = '';
+            angular.element(evt.currentTarget).closest('li')
+              .find('a').blur();
           };
-          $scope.onMouseoutDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
+          /**
+           * Handles keydown events on menus.
+           * @param {object} evt
+           * @param {String} menuName - name of menu to perform action 
+           * on(aboutMenu/profileMenu)
+           * @param {object} eventsTobeHandled - Map keyboard events('Enter') to
+           * corresponding actions to be performed(open/close).
+           * 
+           * @example
+           *  onMenuKeypress($event, 'aboutMenu', {enter: 'open'})
+           */
+          $scope.onMenuKeypress = function(evt, menuName, eventsTobeHandled) {
+            var targetEvents = Object.keys(eventsTobeHandled);
+            for (var i = 0; i < targetEvents.length; i++) {
+              var keyCodeSpec = 
+                $scope.KEYBOARD_EVENT_TO_KEY_CODES[targetEvents[i]];
+              if (keyCodeSpec.keyCode === evt.keyCode && 
+                evt.shiftKey === keyCodeSpec.shiftKeyIsPressed) {
+                if (eventsTobeHandled[targetEvents[i]] === 'open') {
```

Comment:
Pull out 'open', 'close' into named constants in this controller: ACTION_OPEN, ACTION_CLOSE.

Reply:
Done.Regarding the admin navigation page, I will file that as a separate issue. 
Thanks